operatorcert/__init__.py

```diff
@@ -223,3 +238,75 @@ def verify_changed_files_location(
 
     if wrong_changes:
         raise RuntimeError("There are changes in the invalid path")
+
+
+def parse_pr_title(pr_title: str) -> (str, str):
+    """
+    Test, if PR title complies to regex.
+    If yes, extract the Bundle name and version.
+    """
+    # Verify if PR title follows convention- it should contain the operator name Semver regex from semver.org:
+    # https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    semver_regex = (
+        r"(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|["
+        r"1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+("
+        r"?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
+    )
+
+    regex = rf"^operator ([a-zA-Z0-9-]+) \(({semver_regex})\)$"
+    regex_pattern = re.compile(regex)
+
+    if not regex_pattern.match(pr_title):
+        raise ValueError(
+            f"PR title {pr_title} does not follow the regex 'operator <operator_name> (<semver>)"
+        )
+
+    matching = regex_pattern.search(pr_title)
+    bundle_name = matching.group(1)
+    bundle_version = matching.group(2)
+
+    return bundle_name, bundle_version
+
+
+def verify_pr_uniqueness(
+    available_repositories: str, base_pr_url: str, base_pr_bundle_name: str
+) -> None:
+    """
+    List the active Pull Requests in given GitHub repositories.
+    Find Pull Requests for the same Operator Bundle, and error if they exists.
+    """
+    repos = available_repositories.split(",")
+    base_url = "https://api.github.com/repos/redhat-openshift-ecosystem/"
```

Comment:
Can you make this configurable from the params of the pipeline (and script)? It might come in handy should the repos get migrated or for testing against forks.

Reply:
I made the parameter accept org+repo, as it's a simple approach that allows specifying repositories in different organizations.